nrf-hal-common/src/spis.rs

```diff
@@ -0,0 +1,590 @@
+            });
+        }
+
+        if let Some(p) = cipo_pin {
+            spis.psel.miso.write(|w| {
+                unsafe { w.pin().bits(p.pin()) };
+                #[cfg(any(feature = "52833", feature = "52840"))]
+                w.port().bit(p.port().bit());
+                w.connect().connected()
+            });
+        }
+
+        spis.config
+            .modify(|_r, w| w.cpol().bit(Polarity::ActiveHigh.into()));
+        spis.config
+            .modify(|_r, w| w.cpha().bit(Phase::Trailing.into()));
+        spis.enable.write(|w| w.enable().enabled());
+        Self(spis)
+    }
+
+    /// Sets the ´default´ character (character clocked out in case of an ignored transaction).
+    #[inline(always)]
+    pub fn set_default_char(&self, def: u8) -> &Self {
+        self.0.def.write(|w| unsafe { w.def().bits(def) });
+        self
+    }
+
+    /// Sets the over-read character (character sent on over-read of the transmit buffer).
+    #[inline(always)]
+    pub fn set_orc(&self, orc: u8) -> &Self {
+        self.0.orc.write(|w| unsafe { w.orc().bits(orc) });
+        self
+    }
+
+    /// Sets bit order.
+    #[inline(always)]
+    pub fn set_order(&self, order: Order) -> &Self {
+        self.0.config.modify(|_r, w| w.order().bit(order.into()));
+        self
+    }
+
+    /// Sets serial clock (SCK) polarity.
+    #[inline(always)]
+    pub fn set_polarity(&self, polarity: Polarity) -> &Self {
+        self.0.config.modify(|_r, w| w.cpol().bit(polarity.into()));
+        self
+    }
+
+    /// Sets serial clock (SCK) phase.
+    #[inline(always)]
+    pub fn set_phase(&self, phase: Phase) -> &Self {
+        self.0.config.modify(|_r, w| w.cpha().bit(phase.into()));
+        self
+    }
+
+    /// Sets SPI mode.
+    #[inline(always)]
+    pub fn set_mode(&self, mode: Mode) -> &Self {
+        match mode {
+            Mode::Mode0 => {
+                self.set_polarity(Polarity::ActiveHigh);
+                self.set_phase(Phase::Trailing);
+            }
+            Mode::Mode1 => {
+                self.set_polarity(Polarity::ActiveHigh);
+                self.set_phase(Phase::Leading);
+            }
+            Mode::Mode2 => {
+                self.set_polarity(Polarity::ActiveLow);
+                self.set_phase(Phase::Trailing);
+            }
+            Mode::Mode3 => {
+                self.set_polarity(Polarity::ActiveLow);
+                self.set_phase(Phase::Leading);
+            }
+        }
+        self
+    }
+
+    /// Enables the SPIS instance.
+    #[inline(always)]
+    pub fn enable(&self) -> &Self {
+        self.0.enable.write(|w| w.enable().enabled());
+        self
+    }
+
+    /// Disables the SPIS module.
+    #[inline(always)]
+    pub fn disable(&self) -> &Self {
+        self.0.enable.write(|w| w.enable().disabled());
+        self
+    }
+
+    /// Requests acquiring the SPIS semaphore and waits until acquired.
+    #[inline(always)]
+    pub fn acquire(&self) -> &Self {
```

Comment:
We might want a `try_acquire` method that attempts to acquire, but does not block.